code/controllers/subsystem/movement/movement_types.dm

```diff
@@ -412,18 +432,26 @@
 	SIGNAL_HANDLER
 	id = null
 
-//Returns FALSE if the recalculation failed, TRUE otherwise
+///Tries to calculate a new path for this moveloop.
 /datum/move_loop/has_target/jps/proc/recalculate_path()
 	if(!COOLDOWN_FINISHED(src, repath_cooldown))
 		return
 	COOLDOWN_START(src, repath_cooldown, repath_delay)
-	SEND_SIGNAL(src, COMSIG_MOVELOOP_JPS_REPATH)
-	movement_path = get_path_to(moving, target, max_path_length, minimum_distance, id, simulated_only, avoid, skip_first)
+	if(SSpathfinder.pathfind(moving, target, max_path_length, minimum_distance, id, simulated_only, avoid, skip_first, on_finish = on_finish_callback))
+		is_pathing = TRUE
+		SEND_SIGNAL(src, COMSIG_MOVELOOP_JPS_REPATH)
+
+///Called when a path has finished being created
+/datum/move_loop/has_target/jps/proc/OnFinishPathing(list/path)
```

Comment:
use snake_case, it matches the rest of this file and is the norm over (most of) the codebase